### ISSUE TITLE
Running job on SLURM grid

### ISSUE THREAD

**Issue by CRL-CHAR** [alekseyzimin/masurca#349]
I would like to run hybrid assembly using Masurca on the my university's HPC system that utilizes SLURM as its scheduler. I have 2 questions pertaining to this option listed below.

This is what the architecture per 1 node looks like:
```Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                24
On-line CPU(s) list:   0-23
Thread(s) per core:    1
Core(s) per socket:    12
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Gold 6226 CPU @ 2.70GHz
```

Question 1:
**Since each node only has a total of 24 CPUs, is it possible to use multiple nodes in order to scale up the total number of CPUs?**

My abbreviated config file states the following for the GRID parameters:
```USE_GRID=1
GRID_ENGINE=SLURM
GRID_QUEUE=cpu-small
GRID_BATCH_SIZE=500000000
NUM_THREADS=72
```

I state 72 threads because I originally requested 3 nodes, 24 CPU-per-node.

However, when I checked my log file to obtain the batch command execution...
``` Running on the SLURM grid in 3 batches
3 jobs to run
 
To submit SLURM jobs, please run
 
sbatch -D <path_to_working_dir>/mr_pass1 -J create_mega_reads -a 1-3 -n 72 -p cpu-small -N 1 ./create_mega_reads.sh
```
...I noticed that it states 3 jobs to run, corresponding to the 3 batches, but I would be running on only 1 node (-N 1). Referring back to my architecture, my jobs would fail to launch due to the # CPUs exceeding possible resources per node (-n 72).

Question 2:
**Should I state in my config file that NUM_THREADS=24 instead of 72?**

I am happy to provide any additional info, and thanks in advance to the help.

**Comment by CRL-CHAR** [comment]
I'll give a troubleshoot for anyone wanting to run on Masurca job on a cluster running SLURM.

First, take note of the Masurca documentation for NUM_THREADS. This parameter is for # threads/CPUs/tasks **per node**, so set this value to that.

When a job is submitted to the SLURM scheduler via sbatch using -a or --array flags, the number of jobs submitted will be equal to the number of arrays. For example, ```sbatch N-1 -n 24 a=1-6```, will submit 6 jobs each on a node with 24 tasks per node. 

Be sure to specify other flags such as time ```-t```, account name ```-A```, and memory ```-mem```. These are not specified in the generated Masurca sbatch command and may return an error if they are needed for your cluster system. For memory, specifying ```--mem=0``` will partition all the available memory to your job on the node your job is running on.

I will close this comment, and we can consider it resolved.

**Comment by asan-emirsaleh** [comment]
Thank you for sharing your way to resolve this issue. Some thing is still unclear for me. In initial configuration I defined `NUM_THREADS` to 128. But the `create_mega_reads.sh` script that was created for job array submitting, defines `-t 124`, and MaSuRCA pipeline handler suggested using `-n 16` . I have no such configuration in the cluster. So the question is: should I define either `-n 1` for `sbatch` and `-t 16` for script or `-n 16` and `-t 1`, correspondingly? Should I extend SBATCH params to define CPU cores per task etc. I used a compressed parameters configuration, but it yielded nothing in a day f computing.
